src/App.tsx

```diff
@@ -9,38 +9,44 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as postsActions } from './features/postsSlice';
+import { actions as selectedPostActions } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
+  const {
+    items: posts,
+    loaded,
+    hasError,
+  } = useAppSelector(state => state.posts);
+  const { author } = useAppSelector(state => state.author);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const loadUserPosts = useCallback(
+    (userId: number) => {
+      dispatch(postsActions.setLoaded(false));
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+      getUserPosts(userId)
+        .then(userPosts => dispatch(postsActions.setPosts(userPosts)))
+        .catch(() => dispatch(postsActions.setHasError(true)))
+        // We disable the spinner in any case
+        .finally(() => dispatch(postsActions.setLoaded(true)));
```

Comment:
create `asyncThunk` in the slice file and handle promise states there (i.e. loading, error, data on success request)